Sources/_StringProcessing/Regex/Match.swift

```diff
@@ -159,32 +159,23 @@ extension Regex {
 }
 
 @available(SwiftStdlib 5.7, *)
-extension String {
+extension BidirectionalCollection where SubSequence == Substring {
+  /// Match a regex in its entirety.
+  /// - Parameter r: The regex to match against.
```

Comment:
Hmm... potential parameter names could be `regex` or `pattern` or `regexComponent`....

WDYT?

Reply:
@natecook1000